File: solutions/0009-palindrome-number/solution.cpp
// Problem: 9. Palindrome Number
// Difficulty: Easy
// Language: C++
// Runtime: 0 ms
// Memory: 8.5 MB
// Link: https://leetcode.com/problems/palindrome-number/

class Solution {
public:
    bool isPalindrome(int x) {
        if (x<0)
        {
            return false;
        }
        long long temp=x;
        long long reversed=0;
        while (temp!=0)
        {
            int digit=temp%10;
            reversed=reversed*10+digit;
            temp/=10;
        }
        return (x==reversed);
        
    }
};

File: solutions/0009-palindrome-number/solution.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (x<0)
    {
        return false;
    }
    else if (x<10)
    {
        return true;
    }
    else 
    {
        var temp=x;
        var reserved=0;
        while (temp>0)
        {
            var digit=temp%10;
            reserved=reserved*10+digit;
            temp = Math.floor(temp / 10);
        }
        return (reserved==x);
    }
};
